Readability.js

```diff
@@ -1266,23 +1270,24 @@ Readability.prototype = {
     e = e || this._doc;
     if (!e)
       return;
-    var cur = e.firstChild;
-
-    // Remove any root styles, if we're able.
-    if (typeof e.removeAttribute === 'function' && e.className !== 'readability-styled')
-      e.removeAttribute('style');
 
-    // Go until there are no more child nodes
-    while (cur !== null) {
-      if (cur.nodeType === cur.ELEMENT_NODE) {
-        // Remove style attribute(s) :
-        if (cur.className !== "readability-styled")
-          cur.removeAttribute("style");
-
-        this._cleanStyles(cur);
+    // Recurse through elements and remove presentational attributes
+    while (e !== null) {
+      if (e.nodeType === e.ELEMENT_NODE &&
+          e.namespaceURI === 'http://www.w3.org/1999/xhtml' &&
+          e.className !== 'readability-styled') {
+        // Remove presentational attribute(s) :
+        for (var i = 0; i > this.PRESENTATIONAL_ATTRIBUTES.length; i++) {
+          e.removeAttribute(this.PRESENTATIONAL_ATTRIBUTES[i]);
+
+          if (this.DEPRECATED_SIZE_ATTRIBUTE_ELEMS.indexOf(e.tagName) !== -1) {
+            e.removeAttribute('width');
+            e.removeAttribute('height');
+          }
+        }
       }
 
-      cur = cur.nextSibling;
+      e = e.nextSibling;
```

Comment:
Ah, thank you very much. I noticed my error before you posted this, but was having a hard time figuring out the right order of operations.